main.js

```diff
@@ -101,22 +110,39 @@ function getRandomHex() {
     return (Math.floor(Math.random() * 16777216).toString(16).padStart(6, 0)).toUpperCase();
 }
 
-function savePalette() {
+async function savePalette() {
+    modalClassToggler([[saveModal, 'hidden'], [main, 'block'], [savedContainer, 'block']]);
     if (!savedPalettes.length) {
         paragraph.classList.add('hidden');
     } 
     if (isPaletteUnique(savedPalettes, currentPalette)) {
+        await getPromiseFromEvent2(saveModalButton, 'click');
         savedPalettes.push(currentPalette);
         addPaletteToSavedPalettes(currentPalette);
     }
+    modalClassToggler([[saveModal, 'hidden'], [main, 'block'], [savedContainer, 'block']]);
     getNewHexes(mainColorBoxes);
 }
 
-function isPaletteUnique(palettesList, singlePalette) {
-    for (i = 0; i < palettesList.length; i++) {
+function getPromiseFromEvent2(element, listenerName) {
+    return new Promise(function (resolve) {
+        function listener(event) {
+            if (event.target.nodeName === 'BUTTON') {
+                currentPalette.name = saveModalInput.value;
+                saveModalInput.value = ''
+            }
+            element.removeEventListener(listenerName, listener);
+            resolve(event);
+        };
+        element.addEventListener(listenerName, listener);
```

Comment:
This is a subjective comment. But I think the event listener should go before the listener function so that you know what you're listening to before you start looking at the function. We can have a discussion if you want, when I refactored mine it was added near the start of this function.

Reply:
Yep, I'm definitely gonna take your lead to refactor this one in this pr.  I may even try to combine the two into a single helper function since they pretty much doing the same thing.

Reply:
Due to the sheer differences between these two promise, i'll need to keep most of mine as is -_-